Yank/restraints.py

```diff
@@ -1476,7 +1516,9 @@ def determine_missing_parameters(self, thermodynamic_state, sampler_state, topog
                              'restraint parameters...'.format(attempt, MAX_ATTEMPTS))
 
                 # Randomly pick non-collinear atoms.
-                self.restrained_atoms = self._pick_restrained_atoms(sampler_state, topography)
+                restrained_atoms = self._pick_restrained_atoms(sampler_state, topography)
+                self.restrained_receptor_atoms = restrained_atoms[:3]
```

Comment:
The `_pick_restrained_atoms` won't re-pick atoms that have been explicitly given, it basically just give return them as they are.